content/docs/2.13/authentication-providers/aws.md

```diff
@@ -0,0 +1,78 @@
++++
+title = "Pod Identity Webhook for AWS"
++++
+
+[**AWS IRSA Pod Identity Webhook**](https://github.com/aws/amazon-eks-pod-identity-webhook), which is described more in depth [here](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/), allows you to provide the role name using an annotation on a service account associated with your pod.
+
+You can tell KEDA to use AWS Pod Identity Webhook via `podIdentity.provider`.
+
+```yaml
+podIdentity:
+  provider: aws # Optional. Default: none
+  roleArn: <role-arn|workload> # Optional. Default: RoleArn from annotation on service-account.
```

Comment:
What does roleArn mean? Given this is also an option, I think we might need to consider renaming it to `type`?

Reply:
An option could be to have a separate `type` field with `roleArn|workload` (with the default set to `roleArn`). If `workload` is set, `roleArn` is ignored.

Reply:
That is what I was thinking as well

Reply:
agree with the suggestion

Reply:
oh, I misunderstood it :/ 
What about something like `useWorkloadIdentity`? I'm thinking that using workload identity can be ported to Azure podIdentity and it can be useful, so I prefer to not couple it to AWS. (or maybe `useTargetRefIdentity` or so) 
I mean, we could have 2 fields, one `roleArn|clientId` (AWS and Azure respectively) and other `useWorkloadIdentity`.
If `useWorkloadIdentity: true`, we will ignore the other and if it's false, we will take into account the other.
WDYT?

Reply:
useWorkloadIdentity removes our flexibility in case there is a new thing in the future that superceeds workload identity.

I'm also not sure how Azure comes in to play here for an AWS trigger auth resource?

Reply:
Because the spec is shared as it's part of triggerauthentication, so I prefer to use general terms wherever is possible. I mean, `clientId` or `roleArn` are vendor locked, but `provider` is generic. 
If I could use a generic term for using the workload assigned identity, we can use it for all the cloud providers and for user operation is easier IMHO (but I can be overthinking the things)

Reply:
I think the issue with having something "vendor neutral" for this is that it's not terminology that's familiar to someone not already steeped in it. E.g "workload identity" is not something that's immediately apparent to someone who has not used GKE or AKS. To be fair, EKS is the odd one out for calling it IRSA instead of Workload Identity, but still...

Reply:
I think we should standardize as much as we can, but still provide the best end-user experience which means removing cognitive overload and just call it however the provider calls it.

We can use other names internally but I think we should just call it whatever the vendor calls it. A lot of people using "managed identities" don't know the background of how it works or what the project is called, they just want it to work

That's why I think the introduction of `provider: aws` is good! But I'm not a fan of `roleArn: <role-arn|workload>`

Reply:
> But I'm not a fan of roleArn: <role-arn|workload>

roleArn is part of the vendor naming (it's the AWS' for managed identity), so we should use it. My question is, having this scenario, how would you organize the trigger?

- we allow to use the already set roleArn during KEDA installation
- we allow to override the roleArn directly using the TriggerAuthenticationm
- we allow to override the roleArn, getting the value from the scaled workload

Currently, TriggerAuthentication (for azure) allows only the first 2 options, but current AWS auth (getting info also from scaler metadata) supports all of them, and we should support all of them IMO, and eventually allow this extra approach on azure side. 
That's why I'm trying to find the correct way to name the option of using the TrigerAuthentication provide value or the workloads's value. 
Currently, we are already using `identityOwner` as part of AWS Scalers' spec, so maybe we can introduce this concept to the TriggerAuthentication, something like that

Reply:
I think that this will be the most painful point of the new feature... xD

Maybe we could schedule a call for discussing it, or talk about this during the next community standup.

Reply:
When is the next community standup?

Reply:
November 7th at 5PM CEST
You have all the info here: https://keda.sh/community/

Reply:
I've created a new field `identityOwner` with 3 valid values: `keda`, `workload` or unset (same as `keda`) to choose this option